internal/llm/retry_test.go

```
@@ -0,0 +1,515 @@
+
+	primaryFn := func() (*AnalysisResult, error) {
+		primaryCallCount++
+		if primaryCallCount == 2 {
+			cancel()
+		}
+		return nil, fmt.Errorf("gemini API error: %w", genai.APIError{Code: 503, Message: "service unavailable"})
+	}
+
+	fallbackFn := func() (*AnalysisResult, error) {
+		t.Fatal("fallback should not be called when context is canceled")
+		return nil, nil
```

golangci-lint (2.12.2)

[error] return both a `nil` error and an invalid value: use a sentinel error instead

(nilnil)